kafka.cpp

```diff
@@ -191,6 +191,12 @@ void Kafka::applyConfig_Basic(ConfigCategory*& configData)
 		throw exception();
 	}
 
+	if (rd_kafka_conf_set(m_conf, "compression.codec", configData->getValue("compression").c_str(),
+                              errstr, sizeof(errstr)) != RD_KAFKA_CONF_OK)
+	{
+		Logger::getLogger()->fatal(errstr);
```

Comment:
Fatal errors by definition do not allow the code path to continue.  Given the value of compression copies from an enumeration I suspect this only occurs if there is a missing dependency, if indeed it can ever occur.